File: 4. Fourth_Semester/Computer-Science-Workshop-2(CSW-2)/05. Error Handling in Java/Q15.java
import java.util.Scanner;

public class Q15 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try {
            System.out.println("Enter a number: ");
            String input = sc.nextLine();
            try {
                int num = Integer.parseInt(input);
                try {
                    int result = 10 / num;
                    System.out.println("Result of division : "+result);
                } catch (ArithmeticException e) {
                    System.err.println("ArithmeticException Occurred : "+e.getMessage());
                }
            } catch(NumberFormatException e) {
                System.err.println("NumberFormatException occurred: Invalid input!! Please enter a valid integer!");
            }
        } catch(Exception e) {
            System.err.println("An error occurred : "+e.getMessage());
        } finally {
            sc.close();
        }
    }
}

/**
 * OUTPUT
 * Enter a number:
 * 45
 * Result of division : 0
 */

File: 5. Fifth_Semester/PYTHON/Minor Assignment 2/Q15.py
def isPerfectNumber(n):
    sumDivisors = 0
    for i in range(1, n):
        if(n % i == 0):
            sumDivisors += i
    return sumDivisors == n

num = int(input("Enter a Natural number :"))
if(isPerfectNumber(num)):
    print(f"{num} is a Perfect Number")
else:
    print(f"{num} is not a Perfect Number")
